site/src/github.rs

```diff
@@ -673,71 +691,106 @@ async fn categorize_benchmark(
     };
+
+    let benchmark_map = get_benchmark_map(ctxt).await;
+    let (primary, secondary) = split_comparison(comparison, benchmark_map);
+
     const DISAGREEMENT: &str = "If you disagree with this performance assessment, \
     please file an issue in [rust-lang/rustc-perf](https://github.com/rust-lang/rustc-perf/issues/new).";
-    let (summary, direction) = match ComparisonSummary::summarize_comparison(&comparison) {
-        Some(s) if comparison_is_relevant(s.confidence(), is_master_commit) => {
-            let direction = s.direction().unwrap();
-            (s, direction)
-        }
-        Some(_) => {
-            let significant_count = comparison
-                .statistics
-                .iter()
-                .filter(|s| s.is_significant())
-                .count();
-            return (
-                format!(
-                    "This benchmark run did not return any relevant results. {} results were found to be statistically significant but too small to be relevant.\n\n{}",
-                    significant_count,
-                    DISAGREEMENT
-                ),
-                None,
-            );
-        }
-        None => {
-            return (
-                format!(
-                    "This benchmark run did not return any relevant results.\n\n{}",
-                    DISAGREEMENT
-                ),
-                None,
-            );
-        }
+
+    if primary.is_none() && secondary.is_none() {
+        return (
+            format!(
+                "This benchmark run did not return any relevant results.\n\n{}",
+                DISAGREEMENT
+            ),
+            None,
+        );
+    }
+
+    let (primary, secondary) = (
+        primary.unwrap_or_else(|| ComparisonSummary::empty()),
+        secondary.unwrap_or_else(|| ComparisonSummary::empty()),
+    );
+
+    let relevant_summary = if comparison_is_relevant(primary.confidence(), is_master_commit) {
+        &primary
+    } else if comparison_is_relevant(secondary.confidence(), is_master_commit) {
+        &secondary
+    } else {
+        let significant_count = primary.significant_changes() + secondary.significant_changes();
+        return (
+            format!(
+                "This benchmark run did not return any relevant results. {} results were found to be statistically significant but too small to be relevant.\n\n{}",
+                significant_count,
+                DISAGREEMENT
+            ),
+            None,
+        );
     };
 
-    let num_improvements = summary.number_of_improvements();
-    let num_regressions = summary.number_of_regressions();
+    let direction = relevant_summary.direction().unwrap();
+    let num_improvements = relevant_summary.number_of_improvements();
+    let num_regressions = relevant_summary.number_of_regressions();
 
-    let description = match direction {
+    let short_summary = match direction {
```

Comment:
I'm not sure how I feel about the short summary. We're dropping one of the categories (either primary or secondary) and not even mentioning which of the two we're summarizing.

Perhaps for now we can just summarize both:

Summary:
* Primary benchmarks: 🎉 relevant improvements found
* Secondary benchmarks: no relevant changes found

Reply:
Good idea! I'll change it like this.